mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -137,6 +214,46 @@ def close(self) -> None:
             self.image_watcher = None
         self.presenter = None  # type: ignore # Model instance to be destroyed -type can be inconsistent
 
+    def add_mean(self, image_data_obj: Image_Data, image_array: np.ndarray) -> None:
+        if self.roi and (self.roi.left, self.roi.top, self.roi.right, self.roi.bottom) != (0, 0, 0, 0):
```

Comment:
Can this be `if self.roi is not None:`. Shouldn't self.roi either be a valid value or None?

Reply:
The problem is that we want to calculate and add the mean to the spectrum in `Presenter.update_image_list()` but the first mean is calculated before the image is displayed in the image view so the roi is returned as (0,0,0,0), this is to cover that situation

Reply:
Is that because `LiveViewWidget.get_roi()` can return `SensibleROI()`? Would it work for it to return `None` in this case? `LiveViewerWindowModel.roi` is already `SensibleROI | None`, so everywhere need to be able to handle `None`

Reply:
Yes I suppose so, I could change it so it it returns `None` rather than a default `SensibleROI`